platform/security/filter/security-filter-csrf/src/test/groovy/org/codice/ddf/security/filter/csrf/CsrfFilterSpec.groovy

```diff
@@ -52,28 +80,45 @@ class CsrfFilterSpec extends Specification {
-        "/admin/jolokia"           | "https://ddf:123/"    | "https://ddf:123/"    | true
-        "/search/catalog/internal" | "https://ddf:123/"    | null                  | true
-        "/search/catalog/internal" | null                  | "https://ddf:123/"    | true
-        "/search/catalog/internal" | "https://ddf:123/"    | "https://ddf:123/"    | true
-        // Web sockets endpoint requiring same origin but no CSRF header
-        "/search/catalog/ws"       | "https://ddf:123/"    | null                  | true
-        "/search/catalog/ws"       | null                  | "https://ddf:123/"    | true
-        "/search/catalog/ws"       | "https://ddf:123/"    | "https://ddf:123/"    | true
-
+        [requestContext, originHeader, refererHeader, hasCsrfHeader] << [
+                // Non-protected contexts
+                ["/", "/subdirectory"],
+                [null, "", EXTERNAL_SITE, DDF_HTTP, DDF_HTTPS, PROXY_HTTP, PROXY_HTTPS, DDF_BADPORT, PROXY_BADPORT],
+                [null, "", EXTERNAL_SITE, DDF_HTTP, DDF_HTTPS, PROXY_HTTP, PROXY_HTTPS, DDF_BADPORT, PROXY_BADPORT],
+                [true, false]
+        ].combinations() + [
+                // Websockets - same origin OR same referer, with/without CSRF header
+                ["/search/catalog/ws", "/search/catalog/ws/subdirectory"],
+                [DDF_HTTP, DDF_HTTPS, PROXY_HTTP, PROXY_HTTPS],
+                [DDF_HTTP, DDF_HTTPS, PROXY_HTTP, PROXY_HTTPS],
+                [true, false]
+        ].combinations() + [
+                ["/search/catalog/ws", "/search/catalog/ws/subdirectory"],
+                [null, ""],
+                [DDF_HTTP, DDF_HTTPS, PROXY_HTTP, PROXY_HTTPS],
+                [true, false]
+        ].combinations() + [
+                ["/search/catalog/ws", "/search/catalog/ws/subdirectory"],
+                [DDF_HTTP, DDF_HTTPS, PROXY_HTTP, PROXY_HTTPS],
+                [null, ""],
+                [true, false]
+        ].combinations() + [
+                //  Protected Contexts - same origin OR referer, with CSRF header
+                ["/admin/jolokia", "/admin/jolokia/subdirectory", "/search/catalog/internal", "/search/catalog/internal/subdirectory"],
+                [DDF_HTTP, DDF_HTTPS, PROXY_HTTP, PROXY_HTTPS],
+                [DDF_HTTP, DDF_HTTPS, PROXY_HTTP, PROXY_HTTPS],
+                [true]
+        ].combinations() + [
+                ["/admin/jolokia", "/admin/jolokia/subdirectory", "/search/catalog/internal", "/search/catalog/internal/subdirectory"],
+                [DDF_HTTP, DDF_HTTPS, PROXY_HTTP, PROXY_HTTPS],
+                [null, ""],
+                [true]
+        ].combinations() + [
+                ["/admin/jolokia", "/admin/jolokia/subdirectory", "/search/catalog/internal", "/search/catalog/internal/subdirectory"],
+                [null, ""],
+                [DDF_HTTP, DDF_HTTPS, PROXY_HTTP, PROXY_HTTPS],
+                [true]
+        ].combinations()
```

Comment:
*cough* still think this is cleaner ;) *cough* 
```groovy
[requestContext, originHeader, refererHeader, hasCsrfHeader] << [
            // Non-protected contexts
            ["/", "/subdirectory"],
            [null, "", EXTERNAL_SITE, DDF_HTTP, DDF_HTTPS, PROXY_HTTP, PROXY_HTTPS, DDF_BADPORT, PROXY_BADPORT],
            [null, "", EXTERNAL_SITE, DDF_HTTP, DDF_HTTPS, PROXY_HTTP, PROXY_HTTPS, DDF_BADPORT, PROXY_BADPORT],
            [true, false]
        ].combinations() + [
            // Websockets - same origin OR same referer, with/without CSRF header
            ["/search/catalog/ws", "/search/catalog/ws/subdirectory"],
            [DDF_HTTP, DDF_HTTPS, PROXY_HTTP, PROXY_HTTPS, null, ""],
            [DDF_HTTP, DDF_HTTPS, PROXY_HTTP, PROXY_HTTPS, null, ""],
            [true, false]
        ].combinations().findAll{!(it[1] in [null, ""] && it[2] in [null, ""])} + [
            //  Protected Contexts - same origin OR referer, with CSRF header
            ["/admin/jolokia", "/admin/jolokia/subdirectory", "/search/catalog/internal", "/search/catalog/internal/subdirectory"],
            [DDF_HTTP, DDF_HTTPS, PROXY_HTTP, PROXY_HTTPS, null, ""],
            [DDF_HTTP, DDF_HTTPS, PROXY_HTTP, PROXY_HTTPS, null, ""],
            [true]
        ].combinations().findAll{!(it[1] in [null, ""] && it[2] in [null, ""])}
```
alas though you can keep it however you wish